backend/src/main/java/saci/domain/service/KnowledgeService.java

```diff
@@ -27,4 +28,15 @@ public Knowledge createKnowledge(Knowledge knowledge) {
     public List<Knowledge> getKnowledges() {
         return knowledgeRepository.findAll();
     }
+
+    public Knowledge editKnowledge(Long knowledgeId, Knowledge updatedKnowledge) {
+        Knowledge existingKnowledge =
+                knowledgeRepository
+                        .findById(knowledgeId)
+                        .orElseThrow(() -> new NotFoundException("Knowledge not found"));
+
+        existingKnowledge.setName(updatedKnowledge.getName());
```

Comment:
Knowledge doesn't have only the name, use updatedKnowledge itself to update the db.